packages/providers/audit-logs-local/src/index.ts

```diff
@@ -1,43 +1,53 @@
-'use strict';
-
-const auditLogContentType = require('./content-types/audit-log');
-
-const provider = {
-  async register({ strapi }) {
+import type { Strapi } from '@strapi/strapi';
+import { schema as auditLogContentType } from './content-types/audit-log';
+
+interface Event {
+  action: string;
+  date: Date;
+  userId: string | number;
+  payload: Record<string, unknown>;
+}
+
+interface Log extends Omit<Event, 'userId'> {
+  user: string | number;
+}
+
+export = {
+  async register({ strapi }: { strapi: Strapi }) {
     const contentTypes = strapi.container.get('content-types');
     if (!contentTypes.keys().includes('admin::audit-log')) {
       strapi.container.get('content-types').add('admin::', { 'audit-log': auditLogContentType });
     }
 
     // Return the provider object
     return {
-      async saveEvent(event) {
-        // Rewrite userId key to user
-        const auditLog = { ...event, user: event.userId };
-        delete auditLog.userId;
+      async saveEvent(event: Event) {
+        const { userId, ...rest } = event;
+
+        const auditLog: Log = { ...rest, user: userId };
 
         // Save to database
         await strapi.entityService.create('admin::audit-log', { data: auditLog });
 
         return this;
       },
 
-      findMany(query) {
+      findMany(query: Record<string, unknown>) {
```

Comment:
This can be much better typed imo, i think `findPage` has types already we can use to pass the generic of the `admin::audit-log` entity to, this will give better type safety for the query.

Reply:
I agree it's just not the goal of this PR :)